testsuite/compound-action.cc

```diff
@@ -0,0 +1,121 @@
+
+#define BOOST_TEST_DYN_LINK
+#define BOOST_TEST_MODULE libstorage
+
+#include "memory"
+#include "string"
+
+#include <boost/test/unit_test.hpp>
+
+#include "storage/Devicegraph.h"
+#include "storage/Actiongraph.h"
+#include "storage/Storage.h"
+#include "storage/Environment.h"
+#include "storage/Devices/Partition.h"
+
+using namespace std;
+using namespace storage;
+
+
+struct Fixture
+{
+    Fixture() :
+    probed_xml("compound-action-probed.xml"),
+    staging_xml("compound-action-staging.xml")
+    {
+	set_probed();
+    }
+
+    void
+    set_probed()
+    {
+	Environment environment(true, ProbeMode::READ_DEVICEGRAPH, TargetMode::DIRECT);
+	environment.set_devicegraph_filename(probed_xml);
+	
+	storage = shared_ptr<Storage>(new Storage(environment));
+	storage->probe();
+    }
+
+    void
+    set_staging()
+    {
+	auto staging = storage->get_staging();
+	staging->load(staging_xml);
+	staging->check();
+    }
+
+    static const CompoundAction*
+    find_compound_action_by_target(const vector<const CompoundAction*>& actions, const Device* device)
+    {
+	for (auto action : actions)
+	{
+	    if (action->get_target_device() == device)
+		return action;
+	}
+
+	return nullptr;
+    }
+
+    shared_ptr<Storage> storage;
+    string probed_xml;
+    string staging_xml;
+};
+
+
+BOOST_FIXTURE_TEST_SUITE(compound_actions, Fixture)
+
+BOOST_AUTO_TEST_CASE(test_size_without_commit_actions)
+{
+    auto actiongraph = storage->calculate_actiongraph();
+
+    BOOST_CHECK(actiongraph->get_commit_actions().size() == 0);
+
+    BOOST_CHECK(actiongraph->get_compound_actions().size() == 0);
+}
+
+
+BOOST_AUTO_TEST_CASE(test_size_with_commit_actions)
+{
+    set_staging();
+
+    auto actiongraph = storage->calculate_actiongraph();
+
+    BOOST_CHECK(actiongraph->get_commit_actions().size() == 182);
+
+    BOOST_CHECK(actiongraph->get_compound_actions().size() == 74);
```

Comment:
Checking the number of actions and compound actions is not really useful to check the functionality. Please write small tests that check the resulting strings of the compound actions.